packages/core/helper-plugin/src/components/TableProvider/TableProvider.js

```diff
@@ -0,0 +1,379 @@
+          <Th>
+            <BaseCheckbox
+              aria-label={formatMessage({
+                id: 'global.select-all-entries',
+                defaultMessage: 'Select all entries',
+              })}
+              checked={areAllEntriesSelected}
+              indeterminate={isIndeterminate}
+              onChange={handleSelectAll}
+            />
+          </Th>
+        )}
+        {children}
+        {withBulkActions && (
+          <Th>
+            <VisuallyHidden>
+              {formatMessage({
+                id: 'global.actions',
+                defaultMessage: 'Actions',
+              })}
+            </VisuallyHidden>
+          </Th>
+        )}
+      </Tr>
+    </Thead>
+  );
+};
+
+TableHead.defaultProps = {
+  withMainAction: false,
+  withBulkActions: false,
+  rows: [],
+};
+
+TableHead.propTypes = {
+  children: PropTypes.node.isRequired,
+  withMainAction: PropTypes.bool,
+  withBulkActions: PropTypes.bool,
+  rows: PropTypes.array,
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * TableHeaders
+ * -----------------------------------------------------------------------------------------------*/
+
+export const TableHeaders = ({ headers }) => {
+  const { formatMessage } = useIntl();
+  const [{ query }, setQuery] = useQueryParams();
+  const sort = query?.sort || '';
+  const [sortBy, sortOrder] = sort.split(':');
+
+  return headers.map(
+    ({ fieldSchema, name, metadatas: { sortable: isSortable, label, mainField } }) => {
+      let isSorted = sortBy === name;
+      const isUp = sortOrder === 'ASC';
+
+      // relations always have to be sorted by their main field instead of only the
+      // attribute name; sortBy e.g. looks like: &sortBy=attributeName[mainField]:ASC
+      if (fieldSchema?.type === 'relation' && mainField) {
+        isSorted = sortBy === `${name.split('.')[0]}[${mainField.name}]`;
+      }
+
+      const sortLabel = formatMessage(
+        { id: 'components.TableHeader.sort', defaultMessage: 'Sort on {label}' },
+        { label }
+      );
+
+      const handleClickSort = (shouldAllowClick = true) => {
+        if (isSortable && shouldAllowClick) {
+          let nextSort = name;
+
+          // relations always have to be sorted by their main field instead of only the
+          // attribute name; nextSort e.g. looks like: &nextSort=attributeName[mainField]:ASC
+          if (fieldSchema?.type === 'relation' && mainField) {
+            nextSort = `${name.split('.')[0]}[${mainField.name}]`;
+          }
+
+          setQuery({
+            sort: `${nextSort}:${isSorted && sortOrder === 'ASC' ? 'DESC' : 'ASC'}`,
+          });
+        }
+      };
+
+      return (
+        <Th
+          key={name}
+          action={
+            isSorted && (
+              <IconButton
+                label={sortLabel}
+                onClick={handleClickSort}
+                icon={isSorted && <SortIcon isUp={isUp} />}
```

Comment:
```suggestion
                icon={<SortIcon isUp={isUp} />}
```

this will only be rendered if `isSorted` is true right?